.github/workflows/test.yaml

```
@@ -299,7 +299,9 @@ jobs:
     - name: Set up Python
       uses: actions/setup-python@v4
       with:
-        python-version: '${{ env.DEFAULT_PYTHON_VERSION }}'
+        # Changing this will change the benchmark outcomes, this is expected and
+        # unavoidable.
```

Maybe a bit more explicit, for someone in the far future.

But this is minor comment.
We can also keep your comment.

```suggestion
        # When you will need to change this to run the benchmarks
        # on a different Python version,
        # it will also change the benchmark results.
        # This is expected and unavoidable.
        # The PRs that are changing the Python version can
        # be merged by ignoring the benchmark results.
```